library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersDecoration.java

```diff
@@ -5,212 +5,285 @@
+                    }
+                }
+                canvas.save();
+
+                canvas.translate(translationX, translationY);
+                firstHeader.draw(canvas);
+                canvas.restore();
+                mHeaderRects.put(firstPosition, new Rect(translationX, translationY,
+                        translationX + firstHeader.getWidth(), translationY + firstHeader.getHeight()));
+            }
+            for (int i = 1; i < parent.getChildCount(); i++) {
+                int position = parent.getChildPosition(parent.getChildAt(i));
+                if (hasNewHeader(position)) {
+                    // this header is different than the previous, it must be drawn in the correct place
+                    int translationX = 0;
+                    int translationY = 0;
+                    View header = getHeaderView(parent, position);
+                    if (orientation == LinearLayoutManager.VERTICAL) {
+                        translationY = parent.getChildAt(i).getTop() - header.getHeight();
+                    } else {
+                        translationX = parent.getChildAt(i).getLeft() - header.getWidth();
+                    }
+                    canvas.save();
+                    canvas.translate(translationX, translationY);
+                    header.draw(canvas);
+                    canvas.restore();
+                    mHeaderRects.put(position, new Rect(translationX, translationY,
+                            translationX + header.getWidth(), translationY + header.getHeight()));
+                }
+            }
         }
-        canvas.save();
-        canvas.translate(translationX, translationY);
-        firstHeader.draw(canvas);
-        canvas.restore();
-        mHeaderRects.put(firstPosition, new Rect(translationX, translationY,
-            translationX + firstHeader.getWidth(), translationY + firstHeader.getHeight()));
-      }
-      for (int i = 1; i < parent.getChildCount(); i++) {
-        int position = parent.getChildPosition(parent.getChildAt(i));
-        if (hasNewHeader(position)) {
-          // this header is different than the previous, it must be drawn in the correct place
-          int translationX = 0;
-          int translationY = 0;
-          View header = getHeaderView(parent, position);
-          if (orientation == LinearLayoutManager.VERTICAL) {
-            translationY = parent.getChildAt(i).getTop() - header.getHeight();
-          } else {
-            translationX = parent.getChildAt(i).getLeft() - header.getWidth();
-          }
-          canvas.save();
-          canvas.translate(translationX, translationY);
-          header.draw(canvas);
-          canvas.restore();
-          mHeaderRects.put(position, new Rect(translationX, translationY,
-              translationX + header.getWidth(), translationY + header.getHeight()));
+    }
+
+    private void setHeaderIsInline(long headerId, int position) {
```

Comment:
Why do you need separate `inline` and `stacked` methods? It seems simpler to just have `setHeaderStateAndNotify` and pass in the state you want. Using that you could simplify some of the above logic too since different methods wouldn't have to be used for each state.